### DISCUSSION TITLE
The connection does not work

### DISCUSSION THREAD

**Discussion by DEPSTRCZ** [pivpn/pivpn#1801] [Support]
I had PiVpn running once. I tried and it worked (OpenVPN) Yesterday i have noticed that something is wrong. I can init it the connection and handshake happends correctly. There are bytes sent forward and back and even tcpdump shows traffic. Yet it still does not work. I have also tried reconfiguring to Wireguard. But no luck. I have also followed all the diagnostics steps / faq yet no luck. 

- No Firewall
- Using NetworkManager
- Ports forwarded correctly
- OS: Ubuntu 23.10 (GNU/Linux 6.5.0-1008-raspi aarch64)

**Comment by coolapso** [comment]
Are you trying to connect to the VPN Within the same network? That often causes issues similar to what you see, try connecting from an external network, eg. Mobile network. 
if that doesn't work, next logical thing to check is if you are behind a CGNAT ... I know you see traffic going back and forth ... but its hard from me on my side to judge how accurate that is. 

double check what whatsmyip.org reports as being your IP and what the router says its WAN IP is ...

**Comment by DEPSTRCZ** [comment]
I have a static IP. I have a bit of an expirience in this. Been self
hosting stuff on my own for a couple years. This setup has worked with
Pivpn in the past. But does not work on the pi for some reason. Other Port
forwards work without issue.

**Reply by coolapso** [reply]
add here the output of `pivpn -d` , the Iptables rules are useful as well .. 
`iptables -S` and `iptables -t nat -S`  if I'm not around to answer and help will give at least more info for someone to help you.

**Comment by DEPSTRCZ** [comment]
The GH codeblock cant handle it. Here is a paste of all https://pastebin.com/B9L4xC6u

**Reply by coolapso** [reply]
hmmm .... Stop the docker daemon, double check if the docker rules on iptable are gone and try again ... Think its the Docker iptables rules that conflict with pivpn rules .... I think someone ate somepoint managed to fix the rules and still have docker running but can't remember for sure ...

**Reply by DEPSTRCZ** [reply]
Got rid of the rules. Stopped the daemon yet sill no luck.

**Reply by coolapso** [reply]
can you show again the iptables rules please?

**Reply by DEPSTRCZ** [reply]
https://pastebin.com/yyheDiqc
Yes

**Reply by coolapso** [reply]
they're still there and taking precedence over wireguard rules

```
Chain FORWARD (policy ACCEPT)
num  target     prot opt source               destination         
1    DOCKER-USER  all  --  anywhere             anywhere            
2    DOCKER-ISOLATION-STAGE-1  all  --  anywhere             anywhere            
3    ACCEPT     all  --  anywhere             anywhere             ctstate RELATED,ESTABLISHED
4    DOCKER     all  --  anywhere             anywhere            
5    ACCEPT     all  --  anywhere             anywhere            
6    ACCEPT     all  --  anywhere             anywhere            
7    ACCEPT     all  --  anywhere             10.240.3.0/24        ctstate RELATED,ESTABLISHED /* wireguard-forward-rule */
8    ACCEPT     all  --  10.240.3.0/24        anywhere             /* wireguard-forward-rule */
9    ACCEPT     all  --  anywhere             10.146.64.0/24       ctstate RELATED,ESTABLISHED /* wireguard-forward-rule */
10   ACCEPT     all  --  10.146.64.0/24       anywhere             /* wireguard-forward-rule */
11   ACCEPT     all  --  anywhere             10.4.96.0/24         ctstate RELATED,ESTABLISHED /* openvpn-forward-rule */
12   ACCEPT     all  --  10.4.96.0/24         anywhere             /* openvpn-forward-rule */
 
Chain OUTPUT (policy ACCEPT)
num  target     prot opt source               destination         
 
Chain DOCKER (1 references)
num  target     prot opt source               destination         
 
Chain DOCKER-ISOLATION-STAGE-1 (1 references)
num  target     prot opt source               destination         
 
Chain DOCKER-ISOLATION-STAGE-2 (0 references)
num  target     prot opt source               destination         
 
Chain DOCKER-USER (1 references)
num  target     prot opt source               destination   
```

i'd suggest to disable the docker daemon, 
stop it and reboot the system might be the easiest way. 

After booting up double check if the rules are still there. and give it another go.

**Reply by DEPSTRCZ** [reply]
Oh yeah. I am hella blind gimme min

**Reply by DEPSTRCZ** [reply]
```
Chain INPUT (policy ACCEPT)
target     prot opt source               destination         
ACCEPT     udp  --  anywhere             anywhere             udp dpt:12997 /* wireguard-input-rule */
ACCEPT     udp  --  anywhere             anywhere             udp dpt:domain /* pihole-DNS-rule */
ACCEPT     udp  --  anywhere             anywhere             udp dpt:domain /* pihole-DNS-rule */
ACCEPT     udp  --  anywhere             anywhere             udp dpt:domain /* pihole-DNS-rule */
ACCEPT     udp  --  anywhere             anywhere             udp dpt:1297 /* openvpn-input-rule */
ACCEPT     udp  --  anywhere             anywhere             udp dpt:domain /* pihole-DNS-rule */

Chain FORWARD (policy ACCEPT)
target     prot opt source               destination         
ACCEPT     all  --  anywhere             10.240.3.0/24        ctstate RELATED,ESTABLISHED /* wireguard-forward-rule */
ACCEPT     all  --  10.240.3.0/24        anywhere             /* wireguard-forward-rule */
DOCKER     all  --  anywhere             anywhere            
ACCEPT     all  --  anywhere             anywhere            
ACCEPT     all  --  anywhere             anywhere            
ACCEPT     all  --  anywhere             10.146.64.0/24       ctstate RELATED,ESTABLISHED /* wireguard-forward-rule */
ACCEPT     all  --  10.146.64.0/24       anywhere             /* wireguard-forward-rule */
ACCEPT     all  --  anywhere             10.4.96.0/24         ctstate RELATED,ESTABLISHED /* openvpn-forward-rule */
ACCEPT     all  --  10.4.96.0/24         anywhere             /* openvpn-forward-rule */

Chain OUTPUT (policy ACCEPT)
target     prot opt source               destination         

Chain DOCKER (1 references)
target     prot opt source               destination         

Chain DOCKER-ISOLATION-STAGE-1 (0 references)
target     prot opt source               destination         

Chain DOCKER-ISOLATION-STAGE-2 (0 references)
target     prot opt source               destination         

Chain DOCKER-USER (0 references)
target     prot opt source               destination     ```

Still no luck, the docker is fully uninstalled i have tried rebooting and the rules are gone.

**Reply by coolapso** [reply]
what `pivpn -d` says now?

**Reply by TinCanTech** [reply]
I find `iptables-save` to be easier to grok.

**Reply by coolapso** [reply]
I suspect there's something else going on ... I'm tempted to suggest a clean setup, at this point I cannot trust the current state of the system.

**Reply by DEPSTRCZ** [reply]
pivpn -d says all OK... I am not happy with performing a clean install. I will try it in a VM and if that does not shine some light we will see..

**Reply by DEPSTRCZ** [reply]
Oh,,, I was met with an exit with a message: Exiting due to untest OS.. Its the same Os except it is not the Arch version as my pi tho...
Note: I am using NetwrokManager on my Pi.. But that should not make problems

**Comment by DEPSTRCZ** [comment]
So? How i was able to install it on an unsupported system? Is that a bug or?

**Reply by coolapso** [reply]
its not a bug its intended, there are 3 conditions: https://github.com/pivpn/pivpn/blob/master/auto_install/install.sh#L338-L392

1 - Your OS is supported

2 -Your OS is, **maybe** supported, this means that you might be on a distribution or version that is not directly supported but it still might work, for example your case, Ubuntu Mantic, its still not a tested version however its Ubuntu and it most likely works.  In this stage, the information is quite clear, so maybe you should have read it with a bit more attention.  

https://github.com/pivpn/pivpn/blob/master/auto_install/install.sh#L413C1-L418C5

3 - its not supported and the installer will not continue. 

When your OS is supported or maybe supported, if its not on a raspberry pi, then you also get another warning about the IP address configuration not being possible, this is to account for VMS / VPS where IP Settings are typically handled elsewhere.

**Comment by DEPSTRCZ** [comment]
Well the installer tells me that i my version is not supported and i cant continue.. But on the same version but on Arch i am able to install with the warning..

**Comment by DEPSTRCZ** [comment]
In the WG logs i have found that the when a client connects those logs show up.

```
[Sat Jan 13 21:48:21 2024] wireguard: wg0: Receiving handshake initiation from peer 1 (<ip>:1367)
[Sat Jan 13 21:48:21 2024] wireguard: wg0: Sending handshake response to peer 1 (<ip>:1367)
[Sat Jan 13 21:48:21 2024] wireguard: wg0: Keypair 2 created for peer 1
[Sat Jan 13 21:48:21 2024] wireguard: wg0: Packet has unallowed src IP (fe80::619e:a2a4:23b4:e05b) from peer 1 (<ip>:1367)
[Sat Jan 13 21:48:22 2024] wireguard: wg0: Packet has unallowed src IP (fe80::619e:a2a4:23b4:e05b) from peer 1 (<ip>:1367)
[Sat Jan 13 21:48:22 2024] wireguard: wg0: Packet has unallowed src IP (100.116.154.235) from peer 1 (<ip>:1367)
[Sat Jan 13 21:48:24 2024] wireguard: wg0: Packet has unallowed src IP (100.116.154.235) from peer 1 (<ip>:1367)
[Sat Jan 13 21:48:25 2024] wireguard: wg0: Packet has unallowed src IP (fe80::619e:a2a4:23b4:e05b) from peer 1 (<ip>:1367)
[Sat Jan 13 21:48:26 2024] wireguard: wg0: Packet has unallowed src IP (100.116.154.235) from peer 1 (<ip>:1367)
[Sat Jan 13 21:48:28 2024] wireguard: wg0: Packet has unallowed src IP (100.116.154.235) from peer 1 (<ip>:1367)
[Sat Jan 13 21:48:33 2024] wireguard: wg0: Packet has unallowed src IP (fe80::619e:a2a4:23b4:e05b) from peer 1 (<ip>:1367)
[Sat Jan 13 21:48:36 2024] wireguard: wg0: Packet has unallowed src IP (100.116.154.235) from peer 1 (<ip>:1367)
[Sat Jan 13 21:48:40 2024] wireguard: wg0: Sending keepalive packet to peer 1 (<ip>:1367)
[Sat Jan 13 21:48:49 2024] wireguard: wg0: Packet has unallowed src IP (fe80::619e:a2a4:23b4:e05b) from peer 1 (<ip>:1367)
[Sat Jan 13 21:48:52 2024] wireguard: wg0: Sending keepalive packet to peer 1 (<ip>:1367)
[Sat Jan 13 21:48:53 2024] wireguard: wg0: Packet has unallowed src IP (100.116.154.235) from peer 1 (<ip>:1367)
```

**Comment by DEPSTRCZ** [comment]
I have tried couple mroe things. How ever sitll no luck.
